tests/unit/sandbox/formComponents.spec.js

```diff
@@ -0,0 +1,189 @@
+import { mockSandbox, shallowMockSandbox } from './mockSandbox';
+import CCheckboxGroup from '@/sandbox/components/form/CCheckboxGroup';
+import CCodeEditor from '@/sandbox/components/form/CCodeEditor';
+import CCountrySelect from '@/sandbox/components/form/CCountrySelect';
+import CCurrency from '@/sandbox/components/form/CCurrency';
+import CNumber from '@/sandbox/components/form/CNumber';
+import CRadioGroup from '@/sandbox/components/form/CRadioGroup';
+import CSelect from '@/sandbox/components/form/CSelect';
+import CTextInput from '@/sandbox/components/form/CTextInput';
+import CTextarea from '@/sandbox/components/form/CTextarea';
+import CTimepicker from '@/sandbox/components/form/CTimepicker';
+import CToggle from '@/sandbox/components/form/CToggle';
+import { i18n } from '@/sandbox/services/vue-i18n';
+
+describe('Sandbox form components', () => {
+  test('CCodeEditor', async() => {
+    expect.assertions(9);
+
+    /** @type {Wrapper<CCodeEditor>} */
+    const wrapper = mockSandbox(CCodeEditor);
+    const textarea = wrapper.find('textarea');
+
+    const value = JSON.stringify({ data: { code: ['test code'] } });
+
+    textarea.setValue(value);
+    await wrapper.vm.$nextTick();
+    expect(wrapper.emitted().input[0]).toEqual([value]);
+
+    await wrapper.vm.$nextTick();
+    expect(wrapper.find('textarea').element).toHaveValue(
+      '{"data":{"code":["test code"]}}',
+    );
+
+    const { element: codeElement } = wrapper.find('[data-test-id="code"]');
+    const { element: textElement } = wrapper.find('[data-test-id="text"]');
+
+    wrapper.vm.showCode();
+    await wrapper.vm.$nextTick();
+    expect(codeElement).toBeVisible();
+    expect(textElement).not.toBeVisible();
+
+    wrapper.vm.hideCode();
+    await wrapper.vm.$nextTick();
+    expect(codeElement).not.toBeVisible();
+    expect(textElement).toBeVisible();
+
+    wrapper.vm.showCode();
+    await wrapper.vm.$nextTick();
+    expect(codeElement).toBeVisible();
+    expect(textElement).not.toBeVisible();
+
+    textarea.setValue(['invalid data']);
+    await wrapper.vm.$nextTick();
+    expect(wrapper.vm.valid).toBe(false);
+  });
+
+  test('CCountrySelect', async() => {
+    expect.assertions(1);
+    const wrapper = shallowMockSandbox(CCountrySelect);
+
+    wrapper.vm.$i18n.locale = 'nl';
+    wrapper.vm.$i18n.locale = 'en';
```

Comment:
Je zet locale hier naar EN maar je verwacht hieronder NL

Reply:
Nee, ik verwacht dat er in de array van opties de Engelse naam van het land `nl` aanwezig is.

Reply:
Is puur toevallig dat ik hier Nederland heb gekozen, ik kan ook wel een random ander land doen, het is idd verwarrend